src/librustc_middle/mir/mod.rs

```diff
@@ -1131,6 +1131,7 @@ pub enum TerminatorKind<'tcx> {
         /// `true` if this is from a call in HIR rather than from an overloaded
         /// operator. True for overloaded function call.
         from_hir_call: bool,
+        fn_span: Span,
```

Comment:
Needs a doc comment. Also, are we trying to move away from spans inside MIR data structures for incremental? Seems like `InlineAsm` already added some, however.

Reply:
We're storing an almost identical span (`fn_span`, but starting at the beginning of the call chain) in `Terminator.source_info`, so this shouldn't have any effect on incremental compilation. If we end up creating side tables for `Terminator` and `TerminatorKind`, we could move this out at the same time.

Reply:
I wonder if this should be the span in `Terminator.source_info`. Something for a follow-up PR.